src/Microsoft.AspNetCore.Razor.Tools/CompilerHost.cs

```diff
@@ -49,26 +50,25 @@ public override ServerResponse Execute(ServerRequest request, CancellationToken
 
                 var exitCode = 0;
                 var output = string.Empty;
+                var error = string.Empty;
                 var commandArgs = parsed.args.ToArray();
 
-                var writer = ServerLogger.IsLoggingEnabled ? new StringWriter() : TextWriter.Null;
+                var outputWriter = new StringWriter();
+                var errorWriter = new StringWriter();
 
-                var checker = new DefaultExtensionDependencyChecker(Loader, writer, writer);
-                var app = new Application(cancellationToken, Loader, checker, AssemblyReferenceProvider)
-                {
-                    Out = writer,
-                    Error = writer,
-                };
+                var checker = new DefaultExtensionDependencyChecker(Loader, outputWriter, errorWriter);
+                var app = new Application(cancellationToken, Loader, checker, AssemblyReferenceProvider, outputWriter, errorWriter);
 
                 exitCode = app.Execute(commandArgs);
 
-                if (ServerLogger.IsLoggingEnabled)
-                {
-                    output = writer.ToString();
-                    ServerLogger.Log(output);
-                }
+                output = outputWriter.ToString();
+                error = errorWriter.ToString();
```

Comment:
We did work **specifically** to avoid created a string builder and logging lots of output when it doesn't need to go anywhere. I think we should talk more about why this is important.

Reply:
The main problem I am trying to fix here is this. 
Assume there is a compilation error in one of the views. 
It will try to compile it in the server. That will return an error code of -1 because `RazorGenerate` failed. The MsBuild task would fallback to in process compilation and repeat the same process and fail again. (This is the error we are currently seeing in output)
Instead, we could have reported the error from server without having to fallback to the cli. For that I need to create the StringWriter for capturing the `Error` stream atleast. We can talk tomorrow if `Output` stream is really needed.